pkgs/development/libraries/science/math/petsc/default.nix

```diff
@@ -2,27 +2,31 @@
 
 stdenv.mkDerivation rec {
   pname = "petsc";
-  version = "3.13.1";
+  version = "3.13.2";
 
   src = fetchurl {
     url = "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-${version}.tar.gz";
-    sha256 = "0pr604b9pnryl9q0q5arlhs0xdx7wslca0sbz0pzs9qylmz775qp";
+    sha256 = "0yf5lm8dqpbz1yq9nvd5g2iy2w0rgjp9vcj44ba63rwig37j3clp";
   };
 
-  nativeBuildInputs = [ blas gfortran.cc.lib lapack python ];
+  nativeBuildInputs = [ blas gfortran gfortran.cc.lib lapack python ];
 
-  prePatch = stdenv.lib.optionalString stdenv.isDarwin ''
+  prePatch = ''
+    substituteInPlace configure \
+      --replace /bin/sh /usr/bin/python
```

Comment:
Did you really mean to replace the shell with python?

Reply:
Hihi, yes, I do. If upstream does weird things, we have to keep pace, I guess. Here is the current version of the configure script:

```
#!/bin/sh

"true" '''\'
for pyimpl in python3 python python2; do
  which $pyimpl > /dev/null 2>&1  && echo "Using $pyimpl for Python" && exec $pyimpl "$0" "$@"
done
echo "Error! Could not locate python! Please install and have python3, python, or python2 in PATH."
echo "Or use: /path/to/valid/python configure $@"
exit 127
'''

import sys, os

if sys.version_info < (2,6) or (sys.version_info >= (3,0) and sys.version_info < (3,4)):
  print('************************************************************************')
  print('*      Python version 2.6+ or 3.4+ is required to run ./configure      *')
  print('*         Try: "python2.7 ./configure" or "python3 ./configure"        *')
  print('************************************************************************')
  sys.exit(4)

sys.path.insert(0, os.path.abspath('config'))
import configure
configure.petsc_configure([])
````

TL;DR Upstream does some ugly she-py-bang stuff to find a suitable python executor, my change streamlines this process.

Reply:
Oh my ... worth a comment inline in the nix expr w/ your TLDR and/or a link, for future us

Reply:
Sure, I added it.